android/src/main/java/com/brentvatne/exoplayer/FullScreenPlayerView.kt

```diff
@@ -0,0 +1,136 @@
+                }
+            } catch (ex: Exception) {
+                DebugLog.e("ExoPlayer Exception", "Failed to flag FLAG_KEEP_SCREEN_ON on fullscreen.")
+                DebugLog.e("ExoPlayer Exception", ex.toString())
+            }
+        }
+
+        companion object {
+            private const val UPDATE_KEEP_SCREEN_ON_FLAG_MS = 200L
+        }
+    }
+
+    init {
+        setContentView(containerView, generateDefaultLayoutParams())
+    }
+
+    override fun onStart() {
+        super.onStart()
+        parent = exoPlayerView.parent as ViewGroup?
+        parent?.removeView(exoPlayerView)
+        containerView.addView(exoPlayerView, generateDefaultLayoutParams())
+        playerControlView?.let {
+            updateFullscreenButton(playerControlView, true)
+            parent?.removeView(it)
+            containerView.addView(it, generateDefaultLayoutParams())
+        }
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mKeepScreenOnHandler.removeCallbacks(mKeepScreenOnUpdater)
+        containerView.removeView(exoPlayerView)
+        parent?.addView(exoPlayerView, generateDefaultLayoutParams())
+        playerControlView?.let {
+            updateFullscreenButton(playerControlView, false)
+            containerView.removeView(it)
+            parent?.addView(it, generateDefaultLayoutParams())
+        }
+        parent?.requestLayout()
+        parent = null
+    }
+
+    private fun getFullscreenIconResource(isFullscreen: Boolean): Int {
+        return if (isFullscreen) {
+            androidx.media3.ui.R.drawable.exo_icon_fullscreen_exit
+        } else {
+            androidx.media3.ui.R.drawable.exo_icon_fullscreen_enter
+        }
+    }
+
+    private fun updateFullscreenButton(playerControlView: LegacyPlayerControlView, isFullscreen: Boolean) {
+        val imageButton = playerControlView.findViewById<ImageButton?>(com.brentvatne.react.R.id.exo_fullscreen)
+        imageButton?.let {
+            val imgResource = getFullscreenIconResource(isFullscreen)
+            val desc = if (isFullscreen) {
+                context.getString(androidx.media3.ui.R.string.exo_controls_fullscreen_exit_description)
+            } else {
+                context.getString(androidx.media3.ui.R.string.exo_controls_fullscreen_enter_description)
+            }
+            imageButton.setImageResource(imgResource)
+            imageButton.contentDescription = desc
+        }
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        if (reactExoplayerView.preventsDisplaySleepDuringVideoPlayback) {
+            mKeepScreenOnHandler.post(mKeepScreenOnUpdater)
+        }
+        onBackPressedDispatcher.addCallback(onBackPressedCallback)
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        onBackPressedCallback.handleOnBackPressed()
```

Comment:
Is this function call really necessary?

Reply:
@YangJonghun yes, it's necessary because when the user presses the back button, we need to handle this logic. We can move it to the `onStop` function. What do you think?

Reply:
I don't think this is a good implementation.
OnBackPressedDispatcher should behave according to lifecycle of Activity or Fragment.
That's why BackPress Callback currently also works on fullscreen icon clicks.
To implement this properly, it looks like you'll either have to use DialogFragment or use internal Android API. Since this implementation was not purpose of this PR, So it seems like a good idea to revert it.

Reply:
So, should I either use the deprecated function `onBackPressed` or call this logic in the `onStop` function?

Reply:
For now, it seems like a better idea to use onBackPressed, so I suggest it.